packages/hw-app-eth/src/Eth.ts

```diff
@@ -305,137 +271,42 @@ export default class Eth {
         rawTx.copy(buffer, 0, offset, offset + chunkSize);
       }
 
-      toSend.push(buffer);
+      response = await this.transport
+        .send(0xe0, 0x04, first ? 0x00 : 0x80, 0x00, buffer)
+        .catch((e) => {
+          throw remapTransactionRelatedErrors(e);
+        });
+
       offset += chunkSize;
     }
 
-    const provideForContract = async (address) => {
-      const nftInfo = await getNFTInfo(
-        address,
-        chainIdTruncated,
-        this.loadConfig
-      );
-      if (nftInfo) {
-        log(
-          "ethereum",
-          "loading nft info for " +
-            nftInfo.contractAddress +
-            " (" +
-            nftInfo.collectionName +
-            ")"
-        );
-        await provideNFTInformation(this.transport, nftInfo.data);
-      } else {
-        const erc20Info = byContractAddressAndChainId(
-          address,
-          chainIdTruncated
-        );
-        if (erc20Info) {
-          log(
-            "ethereum",
-            "loading erc20token info for " +
-              erc20Info.contractAddress +
-              " (" +
-              erc20Info.ticker +
-              ")"
-          );
-          await provideERC20TokenInformation(this.transport, erc20Info.data);
-        }
-      }
-    };
-
-    if (decodedTx.data.length >= 10) {
-      const selector = decodedTx.data.substring(0, 10);
-      const nftPluginPayload = await loadNftPlugin(
-        decodedTx.to,
-        selector,
-        chainIdTruncated,
-        this.loadConfig
-      );
-
-      if (nftPluginPayload) {
-        await setPlugin(this.transport, nftPluginPayload);
-      } else {
-        const infos = await loadInfosForContractMethod(
-          decodedTx.to,
-          selector,
-          chainIdTruncated,
-          this.loadConfig
-        );
+    const response_byte: number = response[0];
+    let v = "";
 
-        if (infos) {
-          const { plugin, payload, signature, erc20OfInterest, abi } = infos;
+    if (chainId.times(2).plus(35).plus(1).isGreaterThan(255)) {
+      const oneByteChainId = (chainIdTruncated * 2 + 35) % 256;
```

Comment:
Fully trusting you on the buffer voodoo here and below :stuck_out_tongue:

Reply:
I only moved code on this one.